src/nvim/CMakeLists.txt

```diff
@@ -213,6 +213,8 @@ if(CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN)
   list(APPEND gen_cflags "-DEXITFREE")
 endif()
 
+set(CMAKE_C_FLAGS_DEBUG "-DEXITFREE")
```

Comment:
The way this is currently being done means it doesn't show up in `nvim --version`.  Especially since this changes the behavior of the resulting binary, I think it should show up there.

I think a better change would be to move the whole

```cmake
if(CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN)
  list(APPEND gen_cflags "-DEXITFREE")
endif()
```

up to line 205 and change it to

```cmake
if(DEBUG OR CLANG_ASAN_UBSAN OR CLANG_MSAN OR CLANG_TSAN)
  add_definitions(-DEXITFREE)
endif()
```

This should also allow removing the

```cmake
set_property(TARGET nvim APPEND PROPERTY COMPILE_DEFINITIONS EXITFREE)
```

further down in this file (lines 710, 715, 720).

Reply:
Controversial opinion: I'm not the biggest fan of cmake.

Anyway, thanks for the help.